packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -2680,6 +2634,1144 @@ test.serial('writeGranuleFromApi() removes preexisting granule file from postgre
   t.deepEqual(granuleFiles.filter((file) => file.bucket === fakeFile.bucket), []);
 });
 
+test.serial('writeGranulesFromMessage() does not persist file records to Postgres if the workflow status is "running"', async (t) => {
+  // files not persisted due to writeConstraints
+  const {
+    collectionCumulusId,
+    cumulusMessage,
+    executionCumulusId,
+    filePgModel,
+    granuleId,
+    granuleModel,
+    granulePgModel,
+    knex,
+    providerCumulusId,
+  } = t.context;
+
+  cumulusMessage.meta.status = 'running';
+
+  await writeGranulesFromMessage({
+    cumulusMessage,
+    executionCumulusId,
+    providerCumulusId,
+    knex,
+    granuleModel,
+  });
+
+  const granule = await granulePgModel.get(
+    knex,
+    {
+      granule_id: granuleId,
+      collection_cumulus_id: collectionCumulusId,
+    }
+  );
+
+  t.false(
+    await filePgModel.exists(knex, { granule_cumulus_id: granule.cumulus_id })
+  );
+});
+
+test.serial('writeGranulesFromMessage() does not persist file records to Postgres if the workflow status is "queued"', async (t) => {
+  // files not persisted due to writeConstraints
+  const {
+    collectionCumulusId,
+    cumulusMessage,
+    executionCumulusId,
+    filePgModel,
+    granuleId,
+    granuleModel,
+    granulePgModel,
+    knex,
+    providerCumulusId,
+  } = t.context;
+
+  cumulusMessage.meta.status = 'queued';
+
+  await writeGranulesFromMessage({
+    cumulusMessage,
+    executionCumulusId,
+    providerCumulusId,
+    knex,
+    granuleModel,
+  });
+
+  const granule = await granulePgModel.get(
+    knex,
+    {
+      granule_id: granuleId,
+      collection_cumulus_id: collectionCumulusId,
+    }
+  );
+
+  t.false(
+    await filePgModel.exists(knex, { granule_cumulus_id: granule.cumulus_id })
+  );
+});
+
+test.serial('writeGranulesFromMessage() on re-write with the same granule values and files with "completed" status saves granule records to DynamoDB/PostgreSQL/Elasticsearch with updated product volume, expected values, and files', async (t) => {
```

Comment:
Are these two things in conflict: 
`with the same granule values and files with "completed" status`

`with updated product volume, expected values, and files'` ?
 
If the granule is the same, you wouldn't expect updates to the files, right?     Meaning this should read 'without changing product volume, ....' ?

Reply:
So this confused me at first. I copied some of the test setup from other tests using the same 're-write' verbiage, and the product volume ended up changing which for me was unexpected. I realized it was because product volume is one of the values for granules that gets derived from the message payload, instead of just relying on the passed in granule values. I can certainly change it to also pass in the product volume value which matches what will be calculated, but I thought it was somewhat useful/instructive to know that passing in the same info as before will result in an updated product volume. However, I can see that maybe we could use a separate test case to show that some of these values are derived from the message. Do you think I should do that instead?

Reply:
No - I believe I misread the wording here.